lib/Sema/CSApply.cpp

```diff
@@ -285,8 +226,14 @@ Solution::resolveConcreteDeclRef(ValueDecl *decl,
                 const_cast<clang::FunctionTemplateDecl *>(
                     cast<clang::FunctionTemplateDecl>(decl->getClangDecl())),
                 subst);
-    auto newDecl = generateSpecializedCXXFunctionTemplate(
-        decl->getASTContext(), cast<AbstractFunctionDecl>(decl), subst, newFn);
+    llvm::errs() << "NEW FN: "; newFn->dump();
+    // We failed to specialize this function template. The compiler is going to
+    // exit soon. Return something valid in the meantime.
+    if (!newFn)
```

Comment:
Is this condition tested? Should there be an error presented to the user if the compiler is going to exit (is it going to crash or report an error)?

Reply:
This is the exact same thing I removed from `instantiateCXXFunctionTemplate` (that you commented on above). It's the exact same thing, just moved around.